### PR TITLE
chore(deps): update oidc, mongosh, driver

### PR DESCRIPTION
**Pull request by mabaasit**
### Checklist
- [ ] New tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement)

## Motivation and Context

- [ ] Bugfix
- [ ] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Review by addaleax — CHANGES_REQUESTED** [review]
Keep in mind that because we chose to do breaking changes in oidc-plugin and devtools-connect, this cannot land until the corresponding mongosh changes are also published

**Review by addaleax — APPROVED** [review]

File: packages/compass-connections/src/connections-manager.ts [hunk 1]
```diff
@@ -243,6 +244,20 @@ export class ConnectionsManager extends EventEmitter {
         );
       });
 
+      dataService.on?.('oidcAuthFailed', (error) => {
+        this.logger.warn(
+          'ConnectionsManager',
+          mongoLogId(1_001_000_317),
+          'Failed to authenticate with OIDC',
+          error
+        );
```
Comment (addaleax) [inline packages/compass-connections/src/connections-manager.ts:253]: Fwiw, we should already be logging this from oidc-plugin in all cases

**Review by addaleax — APPROVED** [review]

File: packages/compass-connections/src/connections-manager.ts (hunk 1 above)
Comment (mabaasit) [reply packages/compass-connections/src/connections-manager.ts:253]: yeah correct! we do that in the plugin. removing it now

Comment (mabaasit) [reply packages/compass-connections/src/connections-manager.ts]: removed in [1bb4438](https://github.com/mongodb-js/compass/pull/5879/commits/1bb4438fcbcfa1cef64dfeb0fb4232d0b5f63ab5)

**Pull request merged**